### DISCUSSION TITLE
Can I generate a callgraph from VizTracer traces?

### DISCUSSION THREAD

**Discussion by seagrine** [gaogaotiantian/viztracer#453] [General]
I would like to produce a callgraph from a VizTracer trace. Something like pprof does: https://graphviz.org/Gallery/directed/pprof.html

The benefit vis that you see how much time is spent in each function, on aggregate. For example you might easily see that 30% of total time was spent in a particular function, as a percentage of the total time profiled, even if that function was called many different times.

Is such a thing possible with VizTracer?

**Comment by gaogaotiantian** [comment]
One thing that you'll realize with callgraph is - it almost only works for "examples". In reality there will be too many functions and relations and you can never visualize them all in a call graph.

However, an interactive chart, like flamegraph, can do mostly what you described. VizTracer provides such a feature - you can simply use `vizviewer --flamegraph result.json`.

One difference is that flamegraph is a pure tree structure -> each node only has one parent. So for each node, it represents a function with a specific call stack. From my perspective, this is actually better than only the function itself - context matters.
